src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSHostImplementation.cs

```diff
@@ -237,6 +237,8 @@ public static void UninstallWebWorkerInterop()
             {
                 SynchronizationContext.SetSynchronizationContext(syncContext.previousSynchronizationContext);
             }
+            JSProxyContext.CurrentThreadContext = null;
```

Comment:
I wonder if we need to null this out on every thread

Reply:
I hope not. 

`InstallWebWorkerInterop` is the only place we set `CurrentThreadContext` in and it only runs on UI thread or on `JSWebWorker` once. It doesn't propagate on it's own.

`ExecutionContext` does propagate, but the Tasks which it would propagate into should be all resolved by now.